swap/swap_test.go

```diff
@@ -301,6 +307,161 @@ func TestRepeatedBookings(t *testing.T) {
 	verifyBookings(t, swap, append(bookings, mixedBookings...))
 }
 
+func TestNewSwapFailure(t *testing.T) {
+	dir, err := ioutil.TempDir("", "swarmSwap")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(dir)
+	// a simple rpc endpoint for testing dialing
+	ipcEndpoint := path.Join(dir, "TestSwarmSwap.ipc")
+
+	// windows namedpipes are not on filesystem but on NPFS
+	if runtime.GOOS == "windows" {
+		b := make([]byte, 8)
+		rand.Read(b)
+		ipcEndpoint = `\\.\pipe\TestSwarm-` + hex.EncodeToString(b)
+	}
+
+	_, server, err := rpc.StartIPCEndpoint(ipcEndpoint, nil)
+	if err != nil {
+		t.Error(err)
+	}
+	defer server.Stop()
+
+	prvKey, err := crypto.GenerateKey()
+	if err != nil {
+		t.Error(err)
+	}
+
+	type SWAPConfig struct {
```

Comment:
use `testSwapConfig`, or something, it is only used for tests.

Then you can put it outside the function (not really necessary, I just think it's nicer)

Reply:
Agree. Will change in about 30 minutes from now

Reply:
done